### PR TITLE
unixPB: SLES15(SP3) Compatibility Fixes & Updates

### LINKED ISSUE #2670: Bug in ansible unix playbook for SLES15 SP3 (s390x)
- Bug in ansible playbook

Details:

Whilst running the ansible playbook for the new SLES15(SP3) server, a number of minor bugs have been identified : 

1. Python XML Library Requires Installation
2. Update URL For Dev Tools Package Repo ( For all versions of SLES15 )
3. Update download URL For ZLIB pacakge
4. Add missing unix user groups for zeus & jenkins users
5. Update Memcheck.pl ( Nagios )  checksum

### PR DESCRIPTION
**Pull request by steelhead31**
Fixes #2670 

##### Checklist

- [X] commit message has one of the [standard prefixes](https://github.com/adoptium/infrastructure/blob/master/CONTRIBUTING.md#commit-messages)
- [ ] [faq.md](https://github.com/adoptium/infrastructure/blob/master/FAQ.md) updated if appropriate
- [ ] other documentation is changed or added (if applicable)
- [] playbook changes run through [VPC](https://ci.adoptopenjdk.net/view/Tooling/job/VagrantPlaybookCheck/) or [QPC](https://ci.adoptopenjdk.net/view/Tooling/job/QEMUPlaybookCheck/) (if you have access)
- [X] VPC/QPC not applicable for this PR
- [ ] for inventory.yml changes, bastillion/nagios/jenkins updated accordingly

### CODE REVIEW

File: ansible/playbooks/AdoptOpenJDK_Unix_Playbook/roles/Jenkins_User/tasks/main.yml [hunk 1]
```diff
@@ -21,6 +21,14 @@
     - ansible_distribution == "Solaris"
   tags: [jenkins_user, adoptopenjdk]
 
+- name: Create Jenkins  account group for SLES15/s390x
```
Comment (sxa) [inline ansible/playbooks/AdoptOpenJDK_Unix_Playbook/roles/Jenkins_User/tasks/main.yml:24]: Erroneous space character
```suggestion
- name: Create Jenkins account group for SLES15/s390x
```

File: ansible/playbooks/AdoptOpenJDK_Unix_Playbook/roles/Superuser/tasks/main.yml [hunk 2]
```diff
@@ -43,6 +43,15 @@
     - ansible_distribution == "MacOSX"
   tags: [superuser, adoptopenjdk]
 
+- name: Create Superuser account group for SLES15/s390x
+  ansible.builtin.group:
+    name: zeus
+    state: present
+  when:
+    - (ansible_distribution_major_version == "15" and ansible_architecture == "s390x")
```
Comment (sxa) [inline ansible/playbooks/AdoptOpenJDK_Unix_Playbook/roles/Superuser/tasks/main.yml:51]: Is there a reason this is listed as specific to s390x and 15? Would distribution SLES or similar not be ok?

Comment (steelhead31) [reply ansible/playbooks/AdoptOpenJDK_Unix_Playbook/roles/Superuser/tasks/main.yml:51]: I'll make this generic for s390x, I was just being a little too precise, and ensuring my changes only reflected what I'd tested. I'll make this generic, and run a test on the sles12 box.

File: ansible/playbooks/AdoptOpenJDK_Unix_Playbook/roles/Superuser/tasks/main.yml [hunk 3]
```diff
@@ -43,6 +43,15 @@
     - ansible_distribution == "MacOSX"
   tags: [superuser, adoptopenjdk]
 
+- name: Create Superuser account group for SLES15/s390x
```
Comment (sxa) [inline ansible/playbooks/AdoptOpenJDK_Unix_Playbook/roles/Superuser/tasks/main.yml:46]:
```suggestion
- name: Create Superuser account group for SLES15
```

**Comment by sxa** [comment]
Still failing checks so I cannot approve yet :-) (Although I'm surprised this one is failing on the build check and not the linter...)

**Review by sxa — APPROVED** [review]
Checks look good now, my comments have been addressed, so approving :-)

**Pull request merged**